### ISSUE TITLE
RU language  does not work if we define it in options

### ISSUE THREAD

**Issue by ghost** [select2/select2#4451]
Hello! Thank you very much, the best select!!!
I have a one question

if remove this from code
https://monosnap.com/file/ekqT2Q2HakIIwDwWQYG1icFoaYn0or.png

RU language work fine
https://monosnap.com/file/xMYRcBDlutmILqkSAC8ygdVXJ2Zus1.png

else
https://monosnap.com/file/Nw04ymvnCnrSLngpENk0bPie2keWi6.png

how to use the Ru language if i use highlight search word?
https://monosnap.com/file/xKJ5RFmsc3K72c0sEDlHmu9KdcIwTm.png

This example "underline match when searching for data"
http://stackoverflow.com/questions/28958620/using-select2-4-0-0-with-infinite-data-and-filter#29018243

**Comment by viperet** [comment]
I have same problem - changing one message under 'language' resets language chosen by $.fn.select2.defaults.set('language', 'ru');

see https://jsfiddle.net/fm5ee1yp/1/

**Comment by avant1** [comment]
This problem occurs with all non-default languages.
Messages overriding is not usable with them.

When you define options like this:
```javascript
$('selector').select2({
    language: {
        maximumSelected: function(){
            return 'custom message';
        }
    }
});
```

`language` option is not a string or array, and in this case `en` locale would be used as base:
https://github.com/select2/select2/blob/563198a4a2e0eb9f7adbc4775d360db1037a13db/dist/js/select2.js#L4795

This change was introduced as a fix for #3202.

The only workaround for now is to copy all messages definitions from i18n file and override some of them.

**Comment by ilyaskorik** [comment]
No fix?
